dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgDataModel.ts

```diff
@@ -28,54 +28,48 @@ export const lksgDataModel = [
               "Is your head office, administrative headquarters, registered office or subsidiary located in Germany?",
             label: "Head Office in Germany",
             component: "YesNoFormField",
-            required: true,
-            validation: "required",
+            required: false,
           },
           {
             showIf: (): boolean => true,
```

Comment:
In the CreateLksgData.vue component, you do
`v-if="field.showIf(companyAssociatedLksgData.data)"`
The IDE is complaining that `showIf()` does not expect an input argument, which is correct for all subcategories which have `showIf: (): boolean => true`.

Shouldn't we make `showIf` accept an input everywhere, and hardcode a return value `true` for those subcategories, which should always be visible?

Reply:
Implementing this without much boilerplate and not having unused arguments/variables would look like this:
`showIf: (dataModel: LksgData): boolean => !!dataModel`

This gets rid of the error, but is less understandable in my opinion.

Reply:
I think there is a way with better readability:
We could just make the `showIf()` accept an optional param.  Then the IDE should not complain anymore, no matter if a param is passed or not:

![image](https://github.com/d-fine/Dataland/assets/99255296/4afb92e4-3f9e-49bb-b3e8-19c58f7e95c9)

Would this be ok for you?

Reply:
I like the idea